kind.sh

```diff
@@ -56,14 +56,27 @@ main() {
         exit 1
     fi
 
-    local arch
-    case $(uname -m) in
-        i386)           arch="386" ;;
-        i686)           arch="386" ;;
-        x86_64)         arch="amd64" ;;
-        arm|aarch64)    dpkg --print-architecture | grep -q "arm64" && arch="arm64" || arch="arm" ;;
+    local os="linux"
+    case $(uname) in
+        Darwin) os="darwin" ;;
     esac
-    local cache_dir="${RUNNER_TOOL_CACHE}/kind/${version}/${arch}"
+
+    local arch
+    if [[ "$os" == "darwin" ]]; then
+        case $(uname -m) in
+            x86_64)         arch="amd64" ;;
+            arm)            arch="arm64" ;;
```

Comment:
On my Macbook with Apple Silicon, `uname -m` returns `arm64`, not `arm`.

Reply:
ahhhh thanks, i changed it to arm|arm64 to be safe